samples/client/petstore/python-experimental/tests/test_api_exception.py

```diff
@@ -50,7 +51,8 @@ def test_404_error(self):
         except ApiException as e:
             self.assertEqual(e.status, 404)
             self.assertEqual(e.reason, "Not Found")
-            self.checkRegex(e.body, "Pet not found")
+            body = e.body.decode('utf-8') if six.PY3 else e.body
+            self.checkRegex(body, "Pet not found")
```

Comment:
So we used to have e.body decoded here.
Your change as-is is breaking for the python generator because you are changing the type of e.body.
Sorry, I know that in the related issue I suggested putting it in the api client but I didn't know that it wouldbe a breaking change.
Can you move you decoding code into rest.mustache to make this a non-breaking change?

Reply:
@spacether I actually tried to put the decoding in `rest.py` but there's no access to `response_type` there (which is what is used to determine if something should be decoded or not). Would adding `response_type` as a parameter to `call_api` be preferable to this method (I'm assuming both are breaking changes so I'm not sure which is the better spot)?

Reply:
Hmmm, what if we captured the raised exception in api_client:
- decoded the exception body
- re-raised the exception?
Then we have a nonbreaking change.